### PR TITLE
Add S3BucketPublicReadAclRule rule

### PR DESCRIPTION
**Pull request by carleslopez**
Add S3BucketPublicReadAclRule rule. Checks if a bucket has `PublicRead` ACL

### CODE REVIEW

**Review by ocrawford555 — DISMISSED** [review]
LGTM 👍

File: cfripper/rules/s3_public_access.py [hunk 1]
```diff
@@ -105,3 +105,44 @@ def invoke(self, cfmodel: CFModel, extras: Optional[Dict] = None) -> Result:
...
+    REASON = "S3 Bucket {} should not have a public-read acl"
+    RISK_VALUE = RuleRisk.HIGH
+
+    def invoke(self, cfmodel: CFModel, extras: Optional[Dict] = None) -> Result:
+        result = Result()
+        for logical_id, resource in cfmodel.Resources.items():
+            if (
+                resource.Type == "AWS::S3::Bucket"
+                and hasattr(resource, "Properties")
+                and resource.Properties.get("AccessControl") == "PublicRead"
```
Comment (oscarbc96) [inline cfripper/rules/s3_public_access.py:140]: You can use the resources_filtered_by_type method to filter resources if you want:
```suggestion
        for logical_id, resource in cfmodel.resources_filtered_by_type(("AWS::S3::Bucket", )).items():
            if (
                hasattr(resource, "Properties")
                and resource.Properties.get("AccessControl") == "PublicRead"
```

Is it possible to have a S3Bucket without Prooperties 🤔 ?

**Review by oscarbc96 — APPROVED** [review]
🚀 🚀 🚀

File: cfripper/rules/s3_public_access.py (hunk 1 above)
Comment (carleslopez) [reply cfripper/rules/s3_public_access.py:140]: I will say no, but not sure about it 100%

**Review by oscarbc96 — APPROVED** [review]
🚀 🚀 🚀

**Review by ocrawford555 — APPROVED** [review]

**Pull request merged**